File: app/src/main/java/com/dylanc/viewbinding/sample/base/java/BindingViewDelegate.java
package com.dylanc.viewbinding.sample.base.java;

import android.content.Context;
import android.view.ViewGroup;

import androidx.viewbinding.ViewBinding;

import com.drakeet.multitype.ItemViewDelegate;
import com.dylanc.viewbinding.BindingViewHolder;

import org.jetbrains.annotations.NotNull;

public abstract class BindingViewDelegate<T, VB extends ViewBinding> extends ItemViewDelegate<T, BindingViewHolder<VB>> {
  @NotNull
  @Override
  public BindingViewHolder<VB> onCreateViewHolder(@NotNull Context context, @NotNull ViewGroup parent) {
    return BindingViewHolder.createWithGeneric(this, parent);
  }
}

File: app/src/main/java/com/dylanc/viewbinding/sample/base/kotlin/BindingViewDelegate.kt
@file:Suppress("unused")

package com.dylanc.viewbinding.sample.base.kotlin

import android.content.Context
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.drakeet.multitype.ItemViewDelegate
import com.dylanc.viewbinding.BindingViewHolder
import com.dylanc.viewbinding.newBindingViewHolderWithGeneric

abstract class BindingViewDelegate<T, VB : ViewBinding> : ItemViewDelegate<T, BindingViewHolder<VB>>() {

  override fun onCreateViewHolder(context: Context, parent: ViewGroup) =
    newBindingViewHolderWithGeneric<VB>(parent)
}